action/protocol/staking/read_state.go

```diff
@@ -19,7 +19,7 @@ import (
 
 func readStateBuckets(ctx context.Context, sr protocol.StateReader,
```

Comment:
as @yutongp commented, we should change "sr protocol.StateReader" to "csr CandidateStateReader" or "csm CandidateStateManager". This is the key part of the issue. With this kind of changes, you can directly call csr.allBuckets() and etc.

Reply:
Is that to say I don't need to change the param ```StateReader``` of ```ReadState()``` to csm or csr @CoderZhi @yutongp